wrappers/android/tari_crypto.h

```diff
@@ -0,0 +1,132 @@
+#ifndef tari_crypto_h // Include guard, prevents header being included twice, very important
```

Comment:
I do remember cbindgen having an include_guard and other options we could use to do this.

File was included since the JNI (and C++ compilers in general) assumes C++ unless explicitly marked as C (the `extern c` wrapped in the `#ifdef` shortly after this) and is required for it to work.